### DISCUSSION TITLE
Admin app (Android and IOS)

### DISCUSSION THREAD

**Discussion by kosmas1991** [AzuraCast/AzuraCast#7198] [Show and tell]
Hello community,

I created an admin app in dart/flutter for [IOS ](https://apps.apple.com/us/app/azuracast/id6503451097) and [Android ](https://play.google.com/store/apps/details?id=gr.techzombie.azuracastadmin&hl=en_US) using the excellent API @BusterNeece  and team created. Feel free to suggest new features, report bugs or [commit code](https://github.com/kosmas1991/azuracastadmin).  

Also built a radio station [app ](https://play.google.com/store/apps/details?id=gr.techzombie.radiounicorn&hl=en_US) and web interface using dart/flutter for my [radio](https://radiounicorn.eu/). You could modify the code in order to create your application both Android and IOS. Feel also free to ask for directions in case anyone is interested in. 

I am new dart/flutter developer so be kind with my code :-) Also currently working as a developer, so I don't have much time to maintain the projects. 

[Linkedin](https://gr.linkedin.com/in/kosmasgou)

![image](https://github.com/AzuraCast/AzuraCast/assets/72025079/c19d4df7-eaf0-4436-b92a-1bed4b11c917)

**Comment by bombelman** [comment]
Let me be the first to comment and congratulate you.
The app itself is quite responsive. However it takes some time to load data and shows the spinning circles, so not as fast as the web interface. Not sure if this is the app or connection, as on the web-interface there is no such lag it seems.
Tip: The main app icon would be nicer without the black background imo.
Will use the app and provide more feedback soon !

**Comment by BornKillaz** [comment]
Nice. Keep it up.

**Comment by DJMic** [comment]
Congratulations for your app. I've tried your app public page (radio unicorn) on my phone (Android), but sound stop to play after 5 minutes. You should maybe use another way to play sound, like Flutter Radio player : https://pub.dev/packages/flutter_radio_player . I'm not a developper, but I've used this a few years ago when I had develop my radio app.

**Reply by kosmas1991** [reply]
Thanks, but the issue comes from the cheap VPS server I bought for hosting. There is no issue at other more expensive servers. It doesn't bother me as it is not a professional radio site.

**Comment by cheleentano** [comment]
My data loads for a moment and eventually this appears. Android 13
![photo_2024-06-20_20-09-46](https://github.com/AzuraCast/AzuraCast/assets/50516099/ca2a2f22-77d9-4eeb-8ff3-5797956cfcf6)

**Reply by kosmas1991** [reply]
Hello, I think I know what's wrong. Send me your url and a temp API key to debug it for you. kosmasgou@gmail.com Firstly, be sure your server runs lastest stable version of azuracast.

**Comment by pawelzet** [comment]
@kosmas1991  Congratulations! I also wrote my own native iOS app for my radio station:

![image](https://github.com/AzuraCast/AzuraCast/assets/46452256/9e727a18-ea82-4143-9b4f-a2aa9e6c9d1a)


![image](https://github.com/AzuraCast/AzuraCast/assets/46452256/e18d1eb4-523d-4816-88d7-f3e26f681c09)


![image](https://github.com/AzuraCast/AzuraCast/assets/46452256/e113c4ce-3267-4ff8-8f98-fcb6cc7d2383)

Its only on TestFlight for now as Im waiting to get my broadcasting license :)

I have an Apple developer account, so if you want maybe we can cooperate and make native app to be available in the AppStore? Just an idea :)

**Reply by bombelman** [reply]
where are you broadcasting and what license do you need ?

**Reply by pawelzet** [reply]
> where are you broadcasting and what license do you need ?

Sorry,haven't spot your response before. Im broadcasting in Poland and I need a license (royalties for music) to stream an internet radio. Since last year, the App has grown a bit and the radio is ready, it should finally arrive in the AppStore in April/May this year :)
